datetime/datetime.go

```diff
@@ -0,0 +1,150 @@
+package datetime
+
+import (
+	"fmt"
+	"math"
+	"reflect"
+	"time"
+
+	"encoding/binary"
+
+	"gopkg.in/vmihailenco/msgpack.v2"
+)
+
+// Datetime external type
+// Supported since Tarantool 2.10. See more details in issue
+// https://github.com/tarantool/tarantool/issues/5946
+
+const Datetime_extId = 4
+
+const (
+	SEC_LEN       = 8
+	NSEC_LEN      = 4
+	TZ_OFFSET_LEN = 2
+	TZ_INDEX_LEN  = 2
+)
+
+/**
+ * datetime structure keeps number of seconds and
+ * nanoseconds since Unix Epoch.
+ * Time is normalized by UTC, so time-zone offset
+ * is informative only.
+ */
+type EventTime struct {
+	// Seconds since Epoch
+	Seconds int64
+	// Nanoseconds
+	Nsec int32
+	// Timezone offset (in minutes) from UTC
+	TZOffset int16
+	// Olson timezone id
+	TZIndex int16
+}
+
+func encodeDatetime(e *msgpack.Encoder, v reflect.Value) error {
+	tm := v.Interface().(time.Time)
+
+	seconds := tm.Unix()
+	nanosec := tm.UnixNano() - int64(seconds*int64(math.Pow(10, 9)))
```

Comment:
math.Pow ?? really??? then why you use `- *` instead of `%` ?

Either `nanosec := tm.UnixNano() % 100000000` or `nanosec = tm.UnixNano() - seconds * 1000000000` is acceptable.

If you want, you can use `int64(time.Second)` as constant for 1e9.

`nanosec := tm.UnixNano() - seconds * int64(time.Second)` - ah?

Reply:
Thanks for your comments, but I haven't asked you about review. It's a draft, don't waste your time on it.

Reply:
Sources have been updated and comment is outdated.
@funny-falcon, please join to review of the whole patchset, now I'm ready.